debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotChangeEventSource.java

```diff
@@ -356,20 +359,47 @@ private boolean isTableInvalid(P partition, OffsetContext offsetContext) {
         final TableId currentTableId = (TableId) context.currentDataCollectionId().getId();
         currentTable = databaseSchema.tableFor(currentTableId);
         if (currentTable == null) {
-            LOGGER.warn("Schema not found for table '{}', known tables {}", currentTableId, databaseSchema.tableIds());
-            notificationService.incrementalSnapshotNotificationService().notifyTableScanCompleted(context, partition, offsetContext, totalRowsScanned, UNKNOWN_SCHEMA);
-            nextDataCollection(partition, offsetContext);
-            return true;
+            LOGGER.info("Schema not found for table '{}', known tables {}. Will attempt to retrieve this schema",
+                    currentTableId, databaseSchema.tableIds());
+            try {
+                retrieveAndRefreshSchema(currentTableId, partition, offsetContext);
+                currentTable = databaseSchema.tableFor(currentTableId);
+                if (currentTable == null) {
+                    return warnAndSkip(currentTableId, partition, offsetContext, UNKNOWN_SCHEMA, "Schema retrieval failed to populate the schema as expected for {}");
+                }
+            }
+            catch (Exception e) {
+                LOGGER.warn("Failed to retrieve schema for {}", currentTableId, e);
+                return warnAndSkip(currentTableId, partition, offsetContext, UNKNOWN_SCHEMA, "Schema retrieval failed due to an exception for {}");
+            }
         }
         if (chunkQueryBuilder.getQueryColumns(context, currentTable).isEmpty()) {
-            LOGGER.warn("Incremental snapshot for table '{}' skipped because the table has no primary keys", currentTableId);
-            notificationService.incrementalSnapshotNotificationService().notifyTableScanCompleted(context, partition, offsetContext, totalRowsScanned, NO_PRIMARY_KEY);
-            nextDataCollection(partition, offsetContext);
-            return true;
+            return warnAndSkip(currentTableId, partition, offsetContext, NO_PRIMARY_KEY,
+                    "Incremental snapshot for table '{}' skipped because the table has no primary keys");
         }
         return false;
     }
 
+    private boolean warnAndSkip(TableId currentTableId, P partition, OffsetContext offsetContext, TableScanCompletionStatus status, String reason) {
+        LOGGER.warn(reason, currentTableId);
+        notificationService.incrementalSnapshotNotificationService()
+                .notifyTableScanCompleted(context, partition, offsetContext, totalRowsScanned, status); // Ensure conversion to String
+        nextDataCollection(partition, offsetContext);
+        return true;
```

Comment:
I'll put the return value out.

Reply:
Okay cool, I will make that change.  I see there is also a comment there that I must clean up as it relates to something I was experimenting with but didn't do.  

I will make that change once you're finished with the Oracle build as I'm not sure sure if the auto-triggered action overwrites your oracle tests.

Reply:
Yes, let's wait for it.

Reply:
Oracle failures seems not related. Can you please just address this comment?

Reply:
Done and pushed.  Checks running now.